File: PostOrderTraversal/PostOrderTraversal.java
/*
https://www.geeksforgeeks.org/problems/postorder-traversal/1
*/

/*
class Node {
    int data;
    Node left, right;
    Node(int val){
        data = val;
        left = right = null;
    }
}
*/

import java.util.*;

class Solution {
    
    private ArrayList<Integer> resultantPostOrderTraversal;

    public Solution() {
        resultantPostOrderTraversal = new ArrayList<>();
    }
    
    private void depthFirstSearch_forTrees(Node root) {
        if (root == null) return;
        depthFirstSearch_forTrees(root.left);
        depthFirstSearch_forTrees(root.right);
        resultantPostOrderTraversal.add(root.data);
    }
    
    public ArrayList<Integer> postOrder(Node root) {
        resultantPostOrderTraversal.clear();
        depthFirstSearch_forTrees(root);
        return resultantPostOrderTraversal;
    }
}

File: PostOrderTraversal/PostOrderTraversal.cpp
/*
https://www.geeksforgeeks.org/problems/postorder-traversal/1
*/

/*
class Node {
  public:
    int data;
    Node* left;
    Node* right;

    Node(int val) {
        data = val;
        left = NULL;
        right = NULL;
    }
};
*/

class Solution {
  public:
  
    void dfs(Node *root, vector<int>&res){
        if(root==nullptr) return;
        dfs(root->left, res);
        dfs(root->right, res);
        res.push_back(root->data);
    }
  
    vector<int> postOrder(Node* root) {
        vector<int>res;
        dfs(root, res);
        return res;
    }
};